codalab/lib/metadata_util.py

```diff
@@ -78,12 +77,16 @@ def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
         if not new_initial_metadata[spec.key]:
             default = MetadataDefaults.get_default(spec, bundle_subclass, args)
             new_initial_metadata[spec.key] = default
-    initial_metadata = new_initial_metadata
+    return new_initial_metadata
 
-    # If args.edit exists (when doing 'cl edit'), then we want to show
-    # the editor.
+
+def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
```

Comment:
I'd actually remove args from here entirely and do this if statement in bundle_cli where it is more transparent what's going on.